src/Microsoft.Framework.DesignTimeHost/DesignTimeAssemblyLoadContextFactory.cs

```diff
@@ -62,9 +57,12 @@ private IAssemblyLoadContextFactory GetRuntimeFactory(IServiceProvider servicePr
 
                 // Add a cache dependency on restore complete to reevaluate dependencies
                 ctx.Monitor(_namedDependencyProvider.GetNamedDependency(_project.Name + "_Dependencies"));
-
-                return new AssemblyLoadContextFactory(applicationHostContext.ServiceProvider);
+                
+                return applicationHostContext;
             });
+            
+            var factory = new AssemblyLoadContextFactory(appHostContext.ServiceProvider);
```

Comment:
This is fine for now. I guess we don't really need to cache the ProjectLoader and NuGetLoader because they are mostly stateless.